### PR TITLE
feat(outputs.elasticsearch): Allow settings extra headers for elasticsearch output

### LINKED ISSUE #15476: Allow setting custom http headers for elasticsearch output
### Use Case

to ingest data from telegraf to [VictoriaLogs](https://docs.victoriametrics.com/victorialogs/) it's required to remap telegraf fields to fields, which are supported by VictoriaLogs

### Expected behavior

given below config, should allow properly ingest data into victorialogs from `tail` input
```
[[outputs.elasticsearch]]
  urls = ["http://localhost:9428/insert/elasticsearch"]
  timeout = "1m"
  flush_interval = "30s"
  enable_sniffer = false
  health_check_interval = "0s"
  index_name = "device_log-%Y.%m.%d"
  manage_template = false
  template_name = "telegraf"
  overwrite_template = false
  namepass = ["tail"]
  [outputs.elasticsearch.extra_headers]
    "VL-Msg-Field" = "tail.value"
    "VL-Time-Field" = "@timestamp"
    "VL-Stream-Fields" = "measurement_name"
```

### Actual behavior

Currently ingestion from telegraf doesn't create any records in VictoriaLogs

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
## Summary
Allow settings extra headers for elasticsearch output's client

## Checklist
- [x] No AI generated code was used in this PR

## Related issues
resolves #15476

### CODE REVIEW

**Comment by AndrewChubatiuk** [comment]
!signed-cla

File: plugins/outputs/elasticsearch/README.md [hunk 1]
```diff
@@ -323,6 +323,11 @@ to use them.
   ## no pipeline is used for the metric.
   # use_pipeline = "{{es_pipeline}}"
   # default_pipeline = "my_pipeline"
+  #
+  # Custom HTTP headers
+  # To pass extra HTTP headers please define it in a given below section
+  # [outputs.elasticsearch.extra_headers]
+  #    "X-Custom-Header" = "custom-value"
```
Comment (srebhan) [inline plugins/outputs/elasticsearch/README.md:330]: Can we name this just `headers` please? I don't see where this is "extra"...

**Review by srebhan — COMMENTED** [review]
@AndrewChubatiuk thanks for your contribution! The code looks good, just one naming suggestion from my side...

File: plugins/outputs/elasticsearch/README.md (hunk 1 above)
Comment (AndrewChubatiuk) [reply plugins/outputs/elasticsearch/README.md]: just updated

**Review by powersj — APPROVED** [review]
I'm grabbing this for the release. Sven's comment was resolved so considering this approved.

**Pull request merged**